website/dbt-versions.js

```diff
@@ -0,0 +1,19 @@
+exports.versions = [
+    "1.2",
+    "1.0",
+    "0.21",
+    "0.20"
```

Comment:
Copying from Slack:

> I'm not strictly opposed to including v0.20 + v0.21, especially since we already have those modern/legacy tabs for the `--models`/`--select` rename. I wonder if we can do that, while still making clear that only v1.0 and newer are "officially supported" these days? Perhaps by showing a banner when a user selects v0.20 or v0.21? "You're viewing docs for an outdated version of dbt"?

Context: Starting with v1.0, a minor version of dbt will be "officially supported" for 12 months from its initial release. All pre-v1.0 versions are past their official support period. The new Core Versions page (#1061) will also help with clarifying + setting these expectations.

I could see us adding "end of life" dates to these version constants, and then warning people when looking at docs for a version that:
- has an EOL date fast approaching (say, within the next 3 months)
- is already past its official support period

@runleonarun @JKarlavige What do you think of showing that type of banner for _all_ pages? I think it could go a long way toward teaching users about our version policies. I'm happy to kick that further down the road, and say it's out of scope for this PR. Such a banner also ought to link people to the "Understanding dbt Core versions" page, which is soon to exist :)

Reply:
How [Jinja](https://jinja.palletsprojects.com/en/2.11.x/) does this, for instance:
<img width="968" alt="Screenshot 2022-02-21 at 13 15 46" src="https://user-images.githubusercontent.com/13897643/154953577-36bd57de-741b-4071-aad2-657b9d5c05ec.png">

Reply:
> In the meantime, it would be great to sort out the language we would like to use for this unsupported version banner.

I can work on this!

Reply:
The latest two commits add this feature, with the EOD date being set in the `dbt-versions` file.